src/assets/scripts/client/aircraft/AircraftCollection.js

```diff
@@ -0,0 +1,213 @@
+            // this is not using a direct return simply for readability
+            return new AircraftDefinitionModel(aircraftDefinition);
+        });
+
+        return definitionList;
+    }
+
+    /**
+     * Used to build up the appropriate data needed to instantiate an `AircraftInstanceModel`
+     *
+     * @for AircraftCollection
+     * @method _buildAircraftProps
+     * @param spawnModel {SpawnPatternModel}
+     * @return {object}
+     * @private
+     */
+    _buildAircraftProps(spawnModel) {
+        const airlineId = spawnModel.getRandomAirlineForSpawn();
+        const airlineModel = this._airlineCollection.findAirlineById(airlineId);
+        const aircraftDefinition = this._getAircraftDefinitionForAirlineId(airlineModel);
+        const destination = this._findDestinationFromRouteCode(spawnModel);
+        const { fleet } = airlineNameAndFleetHelper([airlineId]);
+        // TODO: move this to `AirlineModel`
+        const callsign = `${_random(0, 999)}`;
+
+        return {
+            callsign,
+            destination,
+            fleet,
+            category: spawnModel.category,
+            airline: airlineModel.icao,
+            altitude: spawnModel.altitude,
+            speed: spawnModel.speed,
+            heading: spawnModel.heading,
+            position: spawnModel.position,
+            icao: aircraftDefinition.icao,
+            model: aircraftDefinition,
+            route: spawnModel.route,
+            waypoints: _get(spawnModel, 'fixes', [])
+        };
+    }
+
+    /**
+     * Given an `airlineId`, find a random aircraft type from the airline.
+     *
+     * @for AircraftCollection
+     * @method _getAircraftDefinitionForAirlineId
+     * @param airlineId {string}
+     * @return aircraftDefinition {AircraftDefinitionModel}
+     * @private
+     */
+    _getAircraftDefinitionForAirlineId(airlineModel) {
+        // TODO: this should be able to handle specific fleets from within an airline
+        // const airlineModel = this._airlineCollection.findAirlineById(airlineId);
+        const aircraftType = airlineModel.getRandomAircraftTypeFromAllFleets();
+        const aircraftDefinition = _find(this.definitionList, { icao: aircraftType });
+
+        if (typeof aircraftDefinition === 'undefined') {
+            console.error(`undefined aircraftDefinition for ${aircraftType}`);
+
+            // recurse through this method if an error is encountered
+            return this._getAircraftDefinitionForAirlineId(airlineModel);
+        }
+
+        return aircraftDefinition;
+    }
+
+    /**
+     * Determines if `destination` is defined and returns route procedure name if it is not
+     *
+     * @for AircraftCollection
+     * @method _findDestinationFromRouteCode
+     * @param destination {string}
+     * @param route {string}
+     * @return {string}
+     * @private
+     */
+    _findDestinationFromRouteCode({ destination, route }) {
```

Comment:
Further down the line, `destination` is used within the `AircraftStripView` to display and aircraft destination.  For departing aircraft, that destination is the `procedureName`.  This method provides an easy way to set the destination for all aircraft.